sdk/keyvault/keyvault-keys/tests/utils/recorder.ts

```diff
@@ -10,7 +8,7 @@ export function isBrowser(): boolean {
 }
 
 export function escapeRegExp(str: string): string {
-  return encodeURIComponent(str).replace(/[\-\[\]{}()*+?.,\\\^$|#\s]/g, "\\$&");
+  return encodeURIComponent(str).replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
```

Comment:
Not sure about this. Can we confirm these are equivalent regex?

Reply:
They should be :) these are standard eslint optimizations. Here are some tests:
```
> "-[^".replace(/[-[^]/g, 'hola,')
'hola,hola,hola'
> "-[^".replace(/[\-\[\^]/g, 'hola,')
'hola,hola,hola'
```